server/datastore/mysql/software.go

```diff
@@ -3177,8 +3177,10 @@ func (ds *Datastore) ListHostSoftware(ctx context.Context, host *fleet.Host, opt
 	}
 	for _, s := range hostSoftwareInstalls {
 		if _, ok := bySoftwareTitleID[s.ID]; !ok {
-			bySoftwareTitleID[s.ID] = s
-		} else {
+			if opts.OnlyAvailableForInstall || opts.IncludeAvailableForInstall {
+				bySoftwareTitleID[s.ID] = s
+			}
+		} else if opts.OnlyAvailableForInstall || opts.IncludeAvailableForInstall {
 			bySoftwareTitleID[s.ID].LastInstallInstalledAt = s.LastInstallInstalledAt
 			bySoftwareTitleID[s.ID].LastInstallInstallUUID = s.LastInstallInstallUUID
 		}
```

Comment:
Do we need to apply the same behavior on pending uninstalls?

Seems like we would want to re-nest things there as well so we don't bother pulling installs/uninstalls if we're just looking at inventory. Which...this is a bit of a chance, but with refetch-on-successful-install-or-uninstall we don't need to lean nearly as hard on install statuses for inventory reasons so this should be fine.

Reply:
I figured pending uninstalls is fine, because we want to return the software until osquery says it's gone. So whether we have a pending uninstall or even a successful uninstall "doesn't matter".